compiler/qsc_passes/src/entry_point.rs

```diff
@@ -44,13 +46,14 @@ pub enum Error {
 pub(super) fn generate_entry_expr(
     package: &mut Package,
     assigner: &mut Assigner,
+    package_type: PackageType,
 ) -> Vec<super::Error> {
     if package.entry.is_some() {
         return vec![];
     }
     let callables = get_callables(package);
 
-    match create_entry_from_callables(assigner, callables) {
+    match create_entry_from_callables(assigner, callables, package_type) {
```

Comment:
I'm surprised to see PackageType wired through in more places now. I had expected PackageType would just go away, and we now just have projects, and in a project you _may_ add an entry point attribute to an operation, and if you try to directly run a project with a) No entrypoint attribute, b)  No 'Main' operation, and c) No entry expression provided (such as in a launch.json or such), then you get a runtime/launch error.

Or is my overly simplified view of the world an unrealistic fantasy :-)

Reply:
(I guess for compilation you need to locate it also)

Reply:
We still need to have the concept internally of "compile and require an entry point, give me a nice error if you can't find one" for run and debug in VS Code and execution in playground/QCOM editors. Everything else can use Lib, but those are still Exe. So the result of this work is the change in user experience and opening the door to removing the setting from the qsharp.json so users never need to know the concept, but the compiler still needs the distinction.

For clarity, we could in a future refactoring change the name of the variables and the enum to indicate the simple thing this represents: is entry point required. But with this proposed change that refactor becomes an internal design question rather than a user experience question.